Can't build the project - some binaryninja-api functions are missing

I'm trying to build the project locally -  I want to expose some C++ methods in python (DebugAdapter, more methods of DebugAdapterType and more)

The `build.md` doesn't seem to be accurate as `cmake` doesn't find anything if executed inside the build dir
Also, when I:
```
cmake -DBN_API_PATH=/home/dzervas/Lab/binaryninja-api -DBN_INSTALL_DIR=/opt/binaryninja -DLLVM_VERSION=14.0.0 ..
make
```

I get the following error:
```
[ 74%] Linking CXX executable debugger_generator
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(binaryview.cpp.o): in function `BinaryNinja::BinaryView::GetFullStringSize(unsigned long, BNStringType)':
binaryview.cpp:(.text+0x11f1b): undefined reference to `BNGetFullStringSize'
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(filemetadata.cpp.o): in function `BinaryNinja::FileMetadata::RevertUndoActions(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
filemetadata.cpp:(.text+0x13b8): undefined reference to `BNRevertUndoActions'
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(binaryninjaapi.cpp.o): in function `BinaryNinja::WorkerEnqueue(std::function<void ()> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
binaryninjaapi.cpp:(.text+0x11a1): undefined reference to `BNWorkerEnqueueNamed'
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(binaryninjaapi.cpp.o): in function `BinaryNinja::WorkerPriorityEnqueue(std::function<void ()> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
binaryninjaapi.cpp:(.text+0x1443): undefined reference to `BNWorkerPriorityEnqueueNamed'
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(binaryninjaapi.cpp.o): in function `BinaryNinja::WorkerInteractiveEnqueue(std::function<void ()> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
binaryninjaapi.cpp:(.text+0x16e5): undefined reference to `BNWorkerInteractiveEnqueueNamed'
/usr/bin/ld: ../../core/api/out/libbinaryninjaapi.a(database.cpp.o): in function `BinaryNinja::Snapshot::GetUndoData()':
database.cpp:(.text+0x1506): undefined reference to `BNGetSnapshotUndoData'
collect2: error: ld returned 1 exit status
make[2]: *** [api/python/CMakeFiles/debugger_generator.dir/build.make:99: api/python/debugger_generator] Error 1
make[1]: *** [CMakeFiles/Makefile2:268: api/python/CMakeFiles/debugger_generator.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

Total guess but I think that it mixes code from the production installed lib and the dev branch of the cloned binaryninja-api

--- a/build.md
+++ b/build.md
@@ -1,6 +1,20 @@
 # Building
 
-- Download LLDB development build for your OS at https://github.com/Vector35/lldb-artifacts/releases. 
+- Update to the latest development build of Binary Ninja 
+  - Follow [this](https://docs.binary.ninja/guide/index.html#support) guide to do so
+
+- Clone `binaryninja-api` and checkout to the corresponding commit used to build the version of Binary Ninja you are running
+  - If you updated to the latest dev build of Binary Ninja, then you can checkout the api repo to the latest dev as well
+  - A more robust approach is to find the file `api_REVISION.txt` in `BN_INSTALLATION_FOLDER/Resources`
+
+```bash
+git clone https://github.com/Vector35/binaryninja-api.git
+cd binaryninja-api
+git checkout dev
+# or git checkout commit_hash
+```
+
+- Download LLDB development build for your OS at https://github.com/Vector35/lldb-artifacts/releases - make sure that the correct LLDB version is downloaded (`grep 'LLVM_VERSION ' core/CMakeLists.txt` can help)
   - Extract the zip archive to `~/libclang`
 
 - Download Qt development build for your OS at https://github.com/Vector35/qt-artifacts/releases.
@@ -8,26 +22,27 @@
 
 - Build the debugger
 
-```
+```bash
 # Get the source
-git clone https://github.com/Vector35/binaryninja-api.git
 git clone https://github.com/Vector35/debugger.git
 
 # Do an out-of-source build
-mkdir build
+mkdir -p build
 cd build
 
 # Build it
-cmake -DBN_API_PATH=../binaryninja-api 
-      -DBN_INSTALL_DIR=/path/to/binaryninja/installation
+cmake -DBN_API_PATH=../binaryninja-api -DBN_INSTALL_DIR=/path/to/binaryninja/installation ..
 
 make
 ```
 
-The build artifacts will be in the folder `out`.
+While the code is compiling, it's a good time to check the [Binary Ninja slack](https://slack.binary.ninja) for any updates!
+
+The build artifacts will be in the folder `out`. You should find two files `libdebuggercore` and `libdebuggerui`, and two folders `python` and `lldb`. 
 
 - Run the debugger
-  - Open Binary Ninja, disable the setting `corePlugins.debugger` so the debugger that comes with Binary Ninja is not loaded 
-  - Copy everything in the `out` folder to the user plugin folder
+  - Open Binary Ninja, disable the setting `corePlugins.debugger` so the debugger that comes with Binary Ninja is not loaded
+  - Close Binary Ninja
+  - Copy everything in the `out` folder to the user plugin folder - `cp -r out/plugins/* ~/.binaryninja/plugins/`
   - Set the environment variable `BN_STANDALONE_DEBUGGER=1`
   - Launch BinaryNinja

